### PR TITLE
Issue/62/intake for consolidate

### PR DESCRIPTION
**Pull request by ellesmith88**
Related to https://github.com/roocs/34e-mngmt/issues/93

Search catalog functionality has been moved from `rook` to here and implemented in consolidate when `use_catalog=True` for a project. Otherwise consolidate behaves the same as before.

Carsten mentioned that he is "misusing" the pywps database for the quick-lookup of the catalog (https://github.com/roocs/34e-mngmt/issues/93). I'm not sure wha this means - all I've done is copy and paste the search part over, so maybe it needs to be changed.

pywps>=4.4.2,<4.5, intake, fsspec<0.9, pandas and aiohttp are new dependencies

when we start using an intake catalog created by the catalog maker: 
```
sdf = df.fillna({"start_time": MIN_DATETIME, "end_time": MAX_DATETIME})
```
will have to be replaced by 
```
sdf = df.replace({"start_time": {"undefined": MIN_DATETIME}})
sdf = df.replace({"end_time": {"undefined": MAX_DATETIME}})
```
this is flagged in the code.

### CODE REVIEW

**Comment by cehbrecht** [comment]
@ellesmith88 I would not like to have a dependency to `pywps` in `daops`. Also the database lookup `DBIntake` needs a postgres database ... this should not be required by `daops`.

I can not offer a prefect solution 😉  

But as an interim step I would prefer to keep the `DBIntake` implementation in rook. So, rook will use the lookup using the postgres database as an accelerator (as it is now in version 0.5.0). `daops` will use the intake catalog directly ... slower ... but our catalogs are still small enough ... should work.

**Review by cehbrecht — CHANGES_REQUESTED** [review]
See inline comments ... avoid pywps and database dependency.

**Comment by ellesmith88** [comment]
@cehbrecht Thanks for the review - sounds good! I'll move things around  😄

**Comment by ellesmith88** [comment]
> See inline comments ... avoid pywps and database dependency.
@cehbrecht I've made these changes - how does it look now?

File: environment.yml [hunk 1]
```diff
@@ -9,9 +9,13 @@ dependencies:
...
   - elasticsearch>=7.9.1
-  - roocs-utils>=0.2.1
+# - roocs-utils>=0.2.1
+  # catalog
+  - intake
+  - fsspec<0.9
+  - aiohttp
   - pip:
-#     - roocs-utils @ git+https://github.com/roocs/roocs-utils.git@master#egg=roocs-utils
+    - roocs-utils @ git+https://github.com/roocs/roocs-utils.git@master#egg=roocs-utils
```
Comment (cehbrecht) [inline environment.yml:20]: I would keep the "pip from git" only in `requirements.txt`.

**Review by cehbrecht — APPROVED** [review]
Looks good to me 😄

**Pull request merged**